cmd/rig/cmd/capsule/instance/shell.go

```diff
@@ -0,0 +1,125 @@
+package instance
+
+import (
+	"errors"
+	"io"
+	"os"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/rigdev/rig-go-api/api/v1/capsule"
+	capsule_cmd "github.com/rigdev/rig/cmd/rig/cmd/capsule"
+	"github.com/spf13/cobra"
+)
+
+func listen(stream *connect.BidiStreamForClient[capsule.ExecuteRequest, capsule.ExecuteResponse]) chan error {
+	doneChan := make(chan error, 1)
+	go func() {
+		for {
+			resp, err := stream.Receive()
+			if err != nil {
+				doneChan <- err
+				return
+			}
+			switch resp.Response.(type) {
+			case *capsule.ExecuteResponse_Stdout:
+				os.Stdout.Write(resp.GetStdout())
+			case *capsule.ExecuteResponse_Stderr:
+				os.Stderr.Write(resp.GetStderr())
+			}
+		}
+	}()
+	return doneChan
+}
+
+func (c Cmd) shell(cmd *cobra.Command, args []string) error {
+	ctx := c.Ctx
+	arg := ""
+	if len(args) > 0 {
+		arg = args[0]
+	}
+
+	instance, err := c.provideInstanceID(ctx, capsule_cmd.CapsuleID, arg)
+	if err != nil {
+		return err
+	}
+
+	start := &capsule.ExecuteRequest{
+		Request: &capsule.ExecuteRequest_Start_{
+			Start: &capsule.ExecuteRequest_Start{
+				CapsuleId:   capsule_cmd.CapsuleID,
+				InstanceId:  instance,
+				Command:     command,
+				Arguments:   arguments,
+				Tty:         tty,
+				Interactive: interactive,
+			},
+		},
+	}
+
+	stream := c.Rig.Capsule().Execute(ctx)
+
+	defer func() {
+		stream.CloseRequest()
+		stream.CloseResponse()
+	}()
+
+	err = stream.Send(start)
+	if errors.Is(err, io.EOF) {
+		_, err := stream.Receive()
+		return err
+	}
+	if err != nil {
+		return err
+	}
+
+	if interactive {
+		outChan := listen(stream)
+		inChan := make(chan error, 1)
+		go func() {
+			for {
+				buf := make([]byte, 4096)
+				n, err := os.Stdin.Read(buf)
```

Comment:
io.EOF is not an error, and we need to send a "stdin" closed event (should similarly result in a io.EOF in the Stream.Write method)